app/controllers/assignments_controller.rb

```diff
@@ -554,25 +554,135 @@ def download_sample_starter_files
+          end
+          assignment.repository_folder = assignment.short_identifier
+          Assignment.transaction do
+            child_assignment = create_assignment_from_zip(zipfile, 'pr-config-files/properties.yml')
+            unless child_assignment.nil?
+              child_assignment.parent_assignment = assignment
+              child_assignment.repository_folder = assignment.repository_folder
+              raise ActiveRecord::Rollback unless child_assignment.save
+            end
+          end
+          raise ActiveRecord::Rollback unless assignment.save
+          redirect_to edit_assignment_path(assignment.id)
+        end
+      end
+    end
   end
 
   private
 
+  # Helper for <upload_config_files> that searches for the <file_name> in <zipfile> that
+  # contains the settings needed to initialize an assignment and uses it to make a new
+  # unsaved assignment
+  def create_assignment_from_zip(zipfile, file_name)
+    # Read properties file first to initialize assignment
+    prop_file = zipfile.glob(file_name).first
+    if prop_file.nil?
+      # Send error message if cannot find properties file of main assignment
+      # Peer review assignments will fail silently
+      if file_name == 'properties.yml'
+        flash_message(:error, I18n.t('upload_errors.cannot_find_file', item: 'properties'))
+      end
+      return nil
+    end
+    prop_content = read_yaml_properties_file(prop_file)
+    # Validate presence of required keys
+    req_keys = [:short_identifier, :due_date, :description]
+    req_keys.each do |key|
+      unless prop_content.key?(key)
+        if file_name == 'properties.yml'
+          flash_message(:error, I18n.t('assignments.no_attribute_in_file', type: '', item: key))
+        else
+          flash_message(:error, I18n.t('assignments.no_attribute_in_file', type: 'peer review', item: key))
+        end
+        return nil
+      end
+    end
+    # Check for uniqueness
+    unless Assignment.find_by(short_identifier: prop_content[:short_identifier]).nil?
+      flash_message(:error, I18n.t('assignments.upload_file_exists',
+                                   item: prop_content[:short_identifier]))
+      return nil
+    end
+    Assignment.new(prop_content)
+  end
+
+  # Reads the yaml file meant for assignment properties and outputs a hash
+  # in the same structure as an assignment
+  def read_yaml_properties_file(file)
+    assignment_attributes = {}
+    assignment_properties = {}
+    base_attr = [:short_identifier, :description, :message,
+                 :due_date, :is_hidden, :show_total, :type]
+    YAML.safe_load(
+      file.get_input_stream.read.encode(Encoding::UTF_8, 'UTF-8'),
+      [Date, Time, Symbol, ActiveSupport::TimeWithZone, ActiveSupport::TimeZone,
+       ActiveSupport::HashWithIndifferentAccess, ActiveSupport::Duration],
+      [],
+      true
+    ).deep_symbolize_keys.each do |attribute_pair|
+      attribute, value = attribute_pair
+      case attribute
```

Comment:
Since you're also deciding the format of the yaml file that gets downloaded, why not just download it in the same format that you want when you upload it instead of reconfiguring the hash here